Doc/library/os.rst

```diff
@@ -5153,14 +5153,19 @@ Miscellaneous System Information
    .. availability:: Unix.
 
 
-.. function:: cpu_count()
+.. function:: cpu_count(*, affinity=False)
 
-   Return the number of CPUs in the system. Returns ``None`` if undetermined.
+   Return the number of logical CPUs in the system. Returns ``None`` if
+   undetermined.
 
-   This number is not equivalent to the number of CPUs the current process can
-   use.  The number of usable CPUs can be obtained with
-   ``len(os.sched_getaffinity(0))``
+   If *affinity* is true, return the number of logical CPUs the current process
+   can use. On error, raise an exception, instead of returning ``None``.
 
+   Linux control groups, *cgroups*, are not taken in account to get the number
```

Comment:
I feel like people wanting such an API may also want cgroups to also be considered when the real question being answered is really _"How parallel am I usefully allowed to be?"_.

Does that need to be separated out into its own `cgroups_cpuset=True` flag so that people could query one or the other or both?  The use cases I have in mind are all around the above question where I'd always want the combination aka `min(logical_cpus, affinity_cores, cgroups_cpuset_cores)`.

Reply:
My PR just fix the documentation to avoid any misunderstading.

> I feel like people wanting such an API may also want cgroups to also be considered

It is discussed in PR #80235. So far, nobody proposes any PR to implement this.

Maybe this PR is a baby step forward :-)

Reply:
Even if we decide to support `cgroup` in the future, I would like to propose not to use the flag name that can represent the implementation detail. If some platform suggests new things do we have to add a new flag for them?